rest/server_context.go

```diff
@@ -455,9 +455,33 @@ func (sc *ServerContext) getOrAddDatabaseFromConfig(ctx context.Context, config
 	return sc._getOrAddDatabaseFromConfig(ctx, config, options)
 }
 
-func GetBucketSpec(ctx context.Context, config *DatabaseConfig, serverConfig *StartupConfig) (spec base.BucketSpec, err error) {
+// GetBucketSpec returns a BucketSpec from a given DatabaseConfig and StartupConfig.
+func GetBucketSpec(ctx context.Context, config *DatabaseConfig, serverConfig *StartupConfig) (*base.BucketSpec, error) {
```

Comment:
What's the benefit in changing to a pointer return for BucketSpec here?

Reply:
I had changed it to a pointer in a previous iteration because the value that it returns is effectively nil and you wouldn't want to use it. It doesn't matter in this case because we check the error value of the function and performing a construction/copy on BucketSpec is rare and cheap.

I restored the previous behavior.